daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_functions/BitFunctions.tdml

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<tdml:testSuite suiteName="SimpleTypes" description="Section 5 - simple types"
+  xmlns:dfdlx="http://www.ogf.org/dfdl/dfdl-1.0/extensions"
+  xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/" xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:math="http://www.w3.org/2005/xpath-functions/math"
+  xmlns:ex="http://example.com"
+  xmlns:home="http://home.com"
+  xmlns:first="http://first.com"
+  defaultRoundTrip="onePass">
+  <tdml:defineSchema elementFormDefault="unqualified" name="BitFunctions">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat"/>
+        <xs:element name="leftInt">
+          <xs:complexType>
+            <xs:sequence dfdl:separator=",">
+              <xs:element name="num" type="xs:int" dfdl:lengthKind="delimited"/>
+              <xs:element name="count" type="xs:int" dfdl:lengthKind="delimited"/>
+              <xs:element name="value" type="xs:int" dfdl:inputValueCalc="{dfdlx:leftShift(../num,../count)}"/>
+            </xs:sequence>
+          </xs:complexType>
+        </xs:element>
+  </tdml:defineSchema>
+  <tdml:parserTestCase name="leftShift_int_01" root="leftInt" model="BitFunctions">
+              <tdml:document>
```

Comment:
The error you are getting is from the unparse. By default tests parse the data, then unparse back to see if the data "round trips" back to the data you started from. In this case, the data will not round trip because nothing reconstructs all that whitespace surrounding the 1, 2. 

That whitespace is all "data" as far as Daffodil and the TDML runner are concerned. Take out all the whitespace so that it is <tdml:document>1,2</tdml:document>, then I think your test will pass. 

Since this error comes from unparsing the infoset back to see if you still get the same output data. That's called roundTrip="onePass" mode.  Because this set of tests is really just testing these new functions you can make all your tests just test parsing only so they just parse and compare the XML, and don't bother unparsing at all. To do that for all the tests, add, to the tdml:testSuite element, the defaultRoundTrip='none' attribute.  One can add roundTrip="none" to individual test cases, but since all your tests will want to be parse only, adding defaultRoundTrip to the test suite is a better choice.

Reply:
Thank you @mbeckerle it passed after removing the white spaces.Let me know what else I need to change before proceeding to other operators.Also for negative test cases do I have to add another .tdml file?

Reply:
Negative tests can go in the same file. 

To make a  negative test, just remove the <tdml:infoset> element and everything inside it, and instead put in an element like this:
```
<tdml:errors>
 <tdml:error>Schema Definition Error</tdml:error>
 <tdml:error>value</tdml:error><!-- name of the element where the error should be --> 
 <tdml:error>keyword</tdml:error><!-- a keyword the error message should have -->
</tdml:errors>
```
That will result in a negative test which will search the diagnostic messages for the strings mentioned. The comparsion is all done case-insensitively. 

I usually create negative tests by starting from this:
```
<tdml:errors>
  <tdml:error>foobar</tdml:error>
</tdml:errors>
```
Then run it and see what the diagnostic messages are. One will be "did not find 'foobar' in any of the diagnostics." but then it will display all the diagnostic messages and you can choose what words/phrases from them make sense to insist upon in the negative test. The goal is to not be overly specific about the wording that the negative test requires, just capture the key things that any good diagnostic about the failure situation would have to mention.